### PR TITLE
Initial commit to supporting basic adaptive icons 

### PR DESCRIPTION
**Pull request by markmooibroek**
Basic implementation to learn from and work towards fully supporting this feature

### CODE REVIEW

**Comment by markmooibroek** [comment]
References #23

File: example/pubspec.yaml [hunk 1]
```diff
@@ -20,6 +20,8 @@ flutter_icons:
   image_path: "assets/images/icon-710x599.png"
   android: true # can specify file name here e.g. "ic_launcher"
   ios: true # can specify file name here e.g. "My-Launcher-Icon"
+  adaptive_icon_background: "#FF7001E3"
+  adaptive_icon_foreground: "assets/images/icon-foreground-432x432.png"
```
[the remark this answers is not in the document]
Comment (markmooibroek) [reply example/pubspec.yaml]: Mi scuzi :-) Added it

**Comment by MarkOSullivan94** [comment]
Thanks for your contribution @markmooibroek! 

I see that the feature is incomplete right now, are you wanting to continue implementing it or want someone else to continue with it?

If you are going to continue with it, I would rather wait until you have a finished, testable version which I could clone and try out locally before merging into the main repo.

**Comment by markmooibroek** [comment]
Sure ill finish it up in a bit, so you can pull and test it out of the box!

**Comment by MarkOSullivan94** [comment]
Many thanks! I'll have a look over your code when I get a chance to see if I can suggest any tweaks.

If you're wanting to make sure your code continues to work in the future, I'm happy for you to add some more unit tests to the project for your feature which will run each time there is a commit to master and ensure that your feature hasn't been broken?

**Comment by markmooibroek** [comment]
@MarkOSullivan94 Try the project locally now. Should be able to:

* generate it 
* if you change the background color or image asset 
* re-run command
* the adaptive icon will update.

**Comment by MarkOSullivan94** [comment]
Also coming from a Java background (well Android development) before doing some Dart with Flutter :smile: 

I might not get a chance to run it locally until later in the week

**Comment by markmooibroek** [comment]
Nice, just let me know if you see some improvements, there are bound te be some :-)

**Comment by MarkOSullivan94** [comment]
Hey @markmooibroek, only getting round to testing your local branch now. The example works which is fantastic, so thank you for that! 

However I did some testing with devices lower than API 26 and it's showing the old launcher icon which we originally had, got me thinking we need to prompt the users or show a warning that unless their min API is 26, that they will need both a default launcher icon and an adaptive icon.

Might be worthwhile putting this information both as a printed message in the console while producing but also in the documentation.

Open to suggestions on what you think would be best if you think something else should be done for this.

**Comment by markmooibroek** [comment]
Good to hear it works on your end as well. Ill add some disclaimers and warnings when making adaptive icons and fail when trying to create only an adaptive icon.

**Comment by markmooibroek** [comment]
Now just an updated README, maybe you can do that part :-)

**Comment by MarkOSullivan94** [comment]
All tests passed on Travis CI: https://travis-ci.org/MarkOSullivan94/flutter_launcher_icons

**Pull request merged**